dpks/interpretation.py

```diff
@@ -300,22 +322,22 @@ def fit(self, X, y, classifier) -> None:
                 X, y, replace=True, n_samples=X.shape[0] * 1, stratify=y, random_state=i
             )
 
-            if isinstance(classifier, Classifier):
-                clf = classifier
-            else:
-                clf = Classifier(classifier=classifier)
-
-            clf.fit(X_train.values, y_train)
+            explainer = FeatureImportance(n_iterations=self.shuffle_iterations, feature_names=self.feature_names, problem_type=self.problem_type)
 
             if self.downsample_background:
                 rus = RandomUnderSampler(random_state=0)
-                X_resampled, _ = rus.fit_resample(X_train, y_train)
-                clf.interpret(X_resampled)
+
+                X_resampled, y_train = rus.fit_resample(X_train, y_train)
+                clf.fit(X_resampled, y_train)
+                explainer.fit(clf, X_resampled)
+
             else:
-                clf.interpret(X_train)
+                clf.fit(X_train, y_train)
+                explainer.fit(clf, X_train)
+
 
             results[f"iteration_{i}_importance"] = pd.Series(
-                clf.mean_importance / clf.mean_importance.max()
+                explainer.global_explanations / explainer.global_explanations.max()
```

Comment:
I guess I never saw this before but why do we take .max() here?

Reply:
This scales all reported importances to the max importance for the column. So they are "normalized" iteration importances. It puts everything between 0-1. The thought was to remove variance based on classifier output just in case the magnitude of the explanations differed between runs for some reason

Reply:
Gotcha